src/auxiliary/UniquePtr.cpp

```diff
@@ -65,6 +65,9 @@ namespace auxiliary
 } // namespace auxiliary
 
 template <typename T>
+#ifdef __HIPCC__
```

Comment:
Just speaking out loud: I wonder if NVHPC (and maybe NVCC and Intel DPC++/SYCL) _could_ have a similar issue. All going well with those rn?

Reply:
```suggestion
#ifdef __HIPCC__  // ROCm 6.2.4 issue, see #1797
```